### ISSUE TITLE
Suggestion: "Local" death sentence

### ISSUE THREAD

**Issue by ziproot** [endless-sky/endless-sky#9781]
### Problem Description

The "death sentence" conversation appears any time an atrocity outfit/cargo is detected upon landing. This is currently not an issue, however, if the transfer slave jobs are fixed such that you can be caught, the Quarg will imprison you for Quarg technology when really all they found were slaves. This is also an issue with plugins that add other atrocity outfits/cargo.

### Related Issue Links

My list of bugs related to pirate jobs on [Discord](https://discord.com/channels/251118043411775489/536900466655887360/1201722778999328778), of which the PR fixing them is on hold due to this issue.

### Desired Solution

"Death sentence" is kept as is, but a new tag "local death sentence" is given which replaces the default death sentence if you are caught with government defined atrocities. This could allow the Quarg death sentence to only trigger upon being caught with Quarg outfits, for example.

### Alternative Approaches

The Quarg death sentence could be made into a mission, but I don't think this is a good idea as that defeats the purpose of government defined atrocities. You could just never be caught carrying slaves, but I also don't think that's a good idea. 

EDIT: Another option is for the death sentence to be modified to be generic, however, this loses the lore aspect of "you wouldn't download a ringworld"

Finally, the jobs could just be removed entirely. There could still be slavery in the background even if you don't help. Considering you can be a pirate without also being a slaver and for obvious reasons this isn't a popular job to take (which is why it has so many bugs), I am fine with just removing these jobs if the devs are, however, that would just hide the problem, and would still make it impossible to have other atrocity outfits/cargo unless they are government specific as well. It is the easiest approach, though.

EDIT: Here is an alternative @Hurleveur suggested that I am fine with:
> I still think that just having the ability to override the atrocity msg with the same syntax as for the illegal one is the way to go.

**Comment by Hurleveur** [comment]
Cant we just add a branch to the quarg atrocity conversation?

**Comment by ziproot** [comment]
Depends on if you can show the default death sentence conversation in a conversation.

**Comment by Hurleveur** [comment]
We wouldnt need to show that one, and even if we did we can just copy it, it isnt worthy of a new feature I think

We already have a branch in the convo, its easy to add others

**Comment by ziproot** [comment]
> We wouldnt need to show that one, and even if we did we can just copy it, it doesnt necessitate a new feature

You would need to manually check if any of the quarg ships/outfits is present, one at a time, then check if any of the slave missions are active, one at a time. The addition of government atrocities was supposed to prevent the former from being required. If that is being used, why not just remove the death sentence and make it a mission instead? That's simpler to implement as you don't additionally have to implement a new death sentence.

**Comment by Hurleveur** [comment]
You cant make said mission trigger when scanning in space
Also it aint that much you gotta look for and you know if its not one its the other

I think a better solution is to have mission based death sentence for what you're describing but then again, no need tbh

**Comment by Hurleveur** [comment]
Just took a look at the code, it looks like its only on the planet that the custom msg is used (lol) which we prob should fix, and also that it wont use the mission msg, which we could change too.

**Comment by xX-Dillinger-Xx** [comment]
Just a related question, should there be a way to recover a useable save file after a death sentence? 
When you die any other way you just hit ESC, reload, and carry on. But, a death sentence can become much harder to recover from if you have gone past 3 pervious saves. This happened to me, involving Quark tech, and I nearly had to start a new pilot. Luckily, I was able to edit my save file manually. Maybe trigger an autosave upon accepting any missions or committing an atrocity that could result in a 'death sentence'. A warning, "_You may not be able to recover from this decision_" might be nice too. 

A side note, **autosaves** need a unique name so we can have more then 1 at a time. It would be nice to go back and replay certain story lines taking a different branch. Currently, the player is sometimes unaware they may be at a fork in the story and don't make a manual save. At that point the option to go back is gone unless they start a new pilot.

**Comment by ziproot** [comment]
For slave missions, abort the mission. For quarg tech, sell or abandon it outside Quarg space.

**Comment by xX-Dillinger-Xx** [comment]
@ziproot 
Yes, I agree, but I didn't know what would happen when I landed on a Quark plant until it was too late. And having the Quark angry at me forever, breaks some storylines. A manual edit was the best and only real option. I was unaware that every Quark everywhere would be angry.  I assumed I could get away with stealing a Quark ship or two, like you can with other factions in the game. Again, a very distinct warning might have been nice. 

IMO slavery missions should be removed. There is so many other things to do, without having very controversial missions in the game. I'm not a social justice warrior but it just feels wrong.

**Comment by Hurleveur** [comment]
I'd prefer punishing players for taking the slave mission, and making them do a moral trip than removing them honestly.
It's there and I think it can make sense gameplay wise as a high risk high reward mission - that said it could be about transporting an AI overlord just as easily gameplay wise.

> Maybe trigger an autosave upon accepting any missions or committing an atrocity that could result in a 'death sentence'. A warning, "You may not be able to recover from this decision" might be nice too.

That could be a bit too many autosaves for the slave missions. We added a warning for tributing already but the slave mission should just have it mentioned its an atrocity. For the Quarg I wouldn't mind a warning when you first board their ship where they go real mad on you.

> And having the Quark angry at me forever, breaks some storylines. A manual edit was the best and only real option

You don't *have* to take the ships and outfits of the most technologically advanced race of the game that also doesnt want to share any of it ^^'.
You already save edited - I do it too when I get Quarg tech, but I don't think the game should account for players screwing themselves up by save editing, in general. In a way, the Quarg imprison you for save editing and robbing their outfits in their face.
The intrusion countermeasures have a warning about this, that said I could see a point made that we should add it to all Quarg outfits in case all smbd did was looting them.

All of that being said we could maybe just delete the last save when all that happens is you die (if there is more than one, ofc) so players dont accidentally reload a game over? (idk if it's saved even)

**Comment by ziproot** [comment]
> We added a warning for tributing already but the slave mission should just have it mentioned its an atrocity

The slave mission already says this:
> Slavery is punishable by death, so this may be a dangerous job to take

> Yes, I agree, but I didn't know what would happen when I landed on a Quark plant (sic) until it was too late
You can reload your save and you will be on a non-Quarg planet. If you then get rid of their stuff the Quarg will be hostile but they won't kill you on the spot if you land on one of their planets. Quarg being hostile is not the same as Quarg instantly killing you for landing on their planet.

> And having the Quark angry at me forever, breaks some storylines. A manual edit was the best and only real option. I was unaware that every Quark everywhere would be angry

If you steal from Merchants, all of human space that isn't a pirate gets mad at you. If you steal from Wanderers, all of the Wanderers get mad at you. If you steal from the Coalition, all of the Coalition gets mad at you. I don't get why this wouldn't be the case for the Quarg.

As for taking ships, if you steal any naturally spawning faction's ship they will get mad at you, unless you increase your reputation with them first, which you can do with the Quarg. For the death sentence mission literally just reload your save and you will be on a non-Quarg planet, then you can get rid of your Quarg tech. Yes, the Quarg will still be hostile, but that's true of all factions in the game. Nobody likes it when you take their stuff. The reason why you got away with stealing the

**Comment by Hurleveur** [comment]
I think they made reputation with merchants by hunting pirates (or mb got deceived by the navy drone mission)
Anyway we are getting slightly carried away.

I still think that just having the ability to override the atrocity msg with the same syntax as for the illegal one is the way to go.

**Comment by xX-Dillinger-Xx** [comment]
> If you steal from Merchants, all of human space that isn't a pirate gets mad at you. If you steal from Wanderers, all of the Wanderers get mad at you. If you steal from the Coalition, all of the Coalition gets mad at you. I don't get why this wouldn't be the case for the Quarg.

Thanks' for the info. I did not know that because I've never done it. I guess I'm too much of a good guy. LOL The only experience I have is stealing a few 'trophy ships' and gunboats from the Republic and some from the Pug. The Republic only stay hostile until you leave the system.  I guess I should start a new pilot and join the dark side.

**Autosaves:** I agree that we don't need autosave for job board missions. That wasn't my point. Let me clarify. As you all know there are several point's in the game where the decisions you make can change the game drastically. Such as, but not limited to, thinking you are a kung fu ninja, joining side A or B, heading down the dark path of pirating, and attacking the Quark. I feel these are a few places where a specifically named Autosave as well as a warning might be nice. Maybe add the warnings to the "help system" so they don't become annoying. 

This stuff kinda only applies to new players. I mentioned this [elsewhere](https://github.com/endless-sky/endless-sky/issues/9627#issuecomment-1937103648), I nearly left out of frustration the first time I played.  As a new player, questions like "What am I _**supposed**_ to do?", "Now, what?", "Where did the storyline go?", and "What happens if I do this or that?"   all lead to confusion and frustration.  This is just my 2 cents, spend it how you want. 
@Hurleveur @ziproot    Sorry for getting so far off topic. I just like to make people think, and this felt related.

**Comment by Hurleveur** [comment]
I dont like handholding the player but feel free to make a PR/issue about adding autosaves in key places
